src/DataTable/index.tsx

```diff
@@ -208,6 +210,9 @@ export const DataTable = <T extends Record<string, unknown>>(
   // If the incoming data changes, override the table data
   useEffect(() => {
     setData(data)
+    if(editingRow){
```

Comment:
@Emmaalara if the table has existing data this is going to add a new row instead of setting the first row to editing. I _think_ we want to set that first row as editing instead of adding new rows. We only want to add a new row on editing if there are no rows present.